frontend/src/api/k8s/secrets.ts

```diff
@@ -202,3 +203,29 @@ export const deleteSecret = (
       opts,
     ),
   );
+
+export const patchSecretWithInferenceServiceOwnerReference = (
+  secret: SecretKind,
+  inferenceService: InferenceServiceKind,
```

Comment:
Future extensibility suggestion: I think we can make the owner ref type more generic, because there is nothing model serving related to the concept of k8s owner references
```suggestion
  ownerReference: K8sResourceCommon & { metadata.name: string },
```
or

```suggestion
  inferenceService: K8sResourceIdentifier & {metadata.name: string },
```

(The `& {metadata.name: string }` is required because those two have `name` as an optional property)